assets/javascripts/modules/ajaxCallbacks.js

```diff
@@ -107,6 +107,113 @@ var ajaxCallbacks = {
       }
     }
   },
+  apiCollaboratorResponse: {
+    callbacks: {
+      success: function(response, $element, data, helpers, targets, container, type) {
+        var email = $element.find('[name=email]').val();
+        var $list = $('[data-collaborator-list]');
+        var $tr = $('<tr></tr>').attr('data-collaborator-row', email);
+        var $name = $('<td></td>').text(email).addClass('table--large');
+        var $permission = $('<td></td>').addClass('table--large text--right hard--right');
+        var $action = $('<td></td>').addClass('text--right hard--right');
+        var $form = $('<form></form>')
+          .attr('action', $list.data('collaborator-remove-url') + email)
+          .attr('data-ajax-submit', true)
+          .attr('data-callback-args', '')
+          .attr('data-callback-name', 'apiCollaboratorRemoveResponse.callbacks')
+          .addClass('form');
+        var $span = $('<span></span>')
+            .text('Server error, please try again')
+            .addClass('error-notification js-remove-error');
+        var $button = $('<button type="submit"></button>')
+            .text('Remove')
+            .attr('data-remove-collaborator-link', email)
+            .addClass('button button--link button--small flush hard--right');
+        var $info = $element.find('.js-info');
+        var $message = $info.find('.alert__message');
+        var emailMessage = 'We have sent an email to <strong>'+ email +'</strong>';
+        var message =  emailMessage + (response.registeredUser ?
+                    ' confirming they have been added to this application.' :
+                    ' inviting them to register with the API Developer Hub. They cannot access the application until they register.');
+
+        if ($element.find('[name=role]:checked').val() === 'ADMINISTRATOR') {
+          $permission.append($('<span class="faded-text">Admin</span>'));
+        }
+
+        // add to the list
+        $form.append($span, $button);
+        $action.append($form);
+        $tr.append($name, $permission, $action);
+        $list.append($tr);
+
+        // register new form
+        $(document.body).trigger('registerNewForm', $form);
```

Comment:
Ok I see what you are doing here. On success you are manually triggering an event to set up listeners on new forms that have been `ajaxed` in.
I was thinking more of using a `delegate` listener to capture any child events. This is usually how content that has been ajaxed in and needs events is handled.